part-1-cli

AlexanderVVasilenko opened the pull request.

AlexanderVVasilenko committed:
done

AlexanderVVasilenko committed:
added structure

AlexanderVVasilenko committed:
done

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from argparse import ArgumentParser
+from operator import add
+
```
add newline here

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+        return [row.strip().split("|") for row in abbreviations_file]
+
+
+# Function to parse the log files and extract relevant information
```
this comment looks okay, you don't need to delete it

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+    return [racer[1] for racer in sorted_racers[:15]], [racer[1] for racer in sorted_racers[15:]]
+
+
+# Function to parse a log line into code and time
```
better to move this comments inside function and wrap it with triple ". Like """Function to parse a log line into code and time"""

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+# Main script
+if __name__ == "__main__":
+    parser = handle_arguments()
+    args = parser.parse_args()
```
readme is needed with explanation how to use it

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+
+
+# Function to print the report
+def print_report(fastest_racers: list[RacerData], bottom_racers: list[RacerData], is_reversed=False) -> None:
+    if not is_reversed:
+        print("Top 15 Racers:")
+        for i, racer_data in enumerate(fastest_racers, start=1):
+            print(f"{i}. {racer_data.name} | {racer_data.team} | {racer_data.lap_time}")
+
+        print("\n" + '-' * 70 + "\n")
+
+        print("Remaining Racers:")
+        for i, racer_data in enumerate(bottom_racers, start=16):
+            print(f"{i}. {racer_data.name} | {racer_data.team} | {racer_data.lap_time}")
+    else:
+        print("Remaining Racers:")
+        for i in range(len(bottom_racers), 0, -1):
+            print(f"{i + 15}. {bottom_racers[i-1].name} | {bottom_racers[i-1].team} | {bottom_racers[i-1].lap_time}")
+
+        print("\n" + "-" * 70 + "\n")
+
+        print("Top 15 Racers:")
+        for i in range(len(fastest_racers), 0, -1):
+            print(f"{i}. {fastest_racers[i-1].name} | {fastest_racers[i-1].team} | {fastest_racers[i-1].lap_time}")
```
code duplicates here. better to create 2 new functions
print_top_15_racers(...,order) and print_remaining_racers(..,order)

levovit commented on report.py:
```diff
@@ -0,0 +1,132 @@
+    return arg_parser
+
+
+def main() -> tuple[list[RacerData], list[RacerData]]:
```
i think you should rename it

AlexanderVVasilenko committed:
report.py: some changes

AlexanderVVasilenko committed:
Merge pull request #3 from AlexanderVVasilenko/part-2

part-2

AlexanderVVasilenko committed:
report.py:
print_remaining_racers(): added
print_top_15_racers(): added
print_report(): changed with using previous 2 functions

README.md: changed cli command for installing package

AlexanderVVasilenko committed:
main(): renamed to initialize_data()

AlexanderVVasilenko committed:
Merge pull request #5 from AlexanderVVasilenko/part-2

Part 2
